pkg/flink/resources.go

```diff
@@ -156,28 +157,47 @@ func (fc *FlinkCluster) updateJobSpec(taskCtx FlinkTaskContext, taskManagerRepli
 		AfterJobCancelled: flinkOp.CleanupActionDeleteCluster,
 	}
 
-	if strings.HasPrefix(out.JarFile, "gs://") {
+	urls := make([]string, len(artifacts))
+	useGcs := true
+	for i, s := range artifacts {
+		urls[i] = s.String()
+		if useGcs && s.Scheme != "gs" {
+			useGcs = false
+		}
+	}
+
+	// XXX(julient) I don't like that this would just silently fail if the condition is not satisfied
+	if useGcs && len(artifacts) > 0 {
```

Comment:
I think we could fail earlier

Reply:
Yes. However I think we need to validate the job definition in an earlier and separate step of the plugin. Ideally we'd also want to perform some validation in the Flink task.

Reply:
Maybe we should address that in a separate PR. There's already a little validation here and there in the plugin (this, task name validation) and we're likely to add more. There should be a specific place for that.